Dart/src/com/jetbrains/lang/dart/psi/impl/DartUriElementBase.java

```diff
@@ -53,6 +57,25 @@ public PsiReference getReference() {
     final String uri = ((DartUriBasedDirective)parent).getUriString();
     final int uriOffset = ((DartUriBasedDirective)parent).getUriStringOffset();
 
+    if (DartResolver.isServerDrivenResolution()) {
+      final PsiFile refPsiFile = getContainingFile();
+      final int refOffset = getTextOffset();
+      final int refLength = getTextLength();
+      final PluginNavigationRegion region = DartResolver.INSTANCE.findRegion(refPsiFile, refOffset, refLength);
+      if (region != null) {
```

Comment:
region is always null for me (SDK 1.12.0-dev.3.1). The problem is that in case of import statement like this 
`import "dart:core" as core;`
IDE reference range is `"dart:core"` only, but navigation region returned by server is `import "dart:core"`.